src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/NotifyListeners.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Component
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.service.NotifyService
+
+@Component
+class NotifyReferralEventListener(
+  @Value("\${interventions.ui.uri}") private val interventionsUIURI: String,
+  private val notifyService: NotifyService,
+): ApplicationListener<ReferralEvent> {
+  override fun onApplicationEvent(event: ReferralEvent){
+    when(event.type) {
+      ReferralEventType.SENT -> {
+        // fixme: is it ok that this class implicitly knows the URL for the sent referral location?
+        val location = UriComponentsBuilder.fromHttpUrl(interventionsUIURI)
+          .path("referrals/{id}/confirmation")
+          .buildAndExpand(event.referral.id)
```

Comment:
I think it's ok for the listener to know the URL as it's deciding what to do with the event (and where it should go).  Alternatively pick it up from a properties file but I still think it's fine.

Reply:
i think including in properties is a good idea 👍